sat/cli/bmccreds/creds_manager.py

```diff
@@ -67,12 +67,26 @@ class BMCCredsManager:
     CHASSIS_BMC_XNAME_REGEX = re.compile(r'x\d+c\d+b\d+$')
     ROUTER_BMC_XNAME_REGEX = re.compile(r'x\d+c\d+r\d+b\d+$')
 
-    def __init__(self, password, xnames, domain, force, report_format):
+    def __init__(self, password, xnames, domain, force, report_format, length=RANDOM_PASSWORD_LENGTH,
+                 allowed_chars=VALID_CHAR_SETS_STRING):
+        """
+        Initialize the BMCCredsManager instance
+        Args:
+            password (str): The password to use
+            xnames (list of str): A list of xnames
+            domain (str): The domain such as bmc, chassis, cabinets, systems
+            force (bool): value of the 'force' parameter to the SCSD API request
+            report_format (str): the format to print the report either 'pretty', 'yaml', or 'json'
+            length (int):Length of the random password (default: RANDOM_PASSWORD_LENGTH)
+            allowed_chars (str): allowed characters for the random password (default: alpha, numeric and symbols)
```

Comment:
Oh, I just noticed that the arguments to this constructor are all also documented already in the docstring on the class on lines 50-65. If you'd like, you could just add `length` and `allow_chars` there if you'd like. That would reduce the duplication.

Reply:
Removed the docstring from `__init__` constructor to reduce duplication